lib/utils/deprecation/config_file.rb

```diff
@@ -0,0 +1,108 @@
+    class ConfigFile
+      GroupEntry = Struct.new(:name, :action, :prefix, :suffix, :exit_status, :occurences)
+
+      # What actions may you specify to be taken when a deprecation is encountered?
+      VALID_ACTIONS = [
+        # Hard exit inspec, no stacktrace, exit code specified or DEFAULT_HARD_EXIT_STATUS
+        :exit,
+        # Fail the control with the message. If not in a control, do :warn action instead.
+        :fail_control,
+        # Do nothing.
+        :ignore,
+        # Issue a warning
+        :warn,
+      ].freeze
+
+      VALID_GROUP_FIELDS = %w{action suffix prefix exit_status}.freeze
+
+      attr_reader :groups, :unknown_group_action
+
+      def initialize(io = nil)
+        io ||= open_default_config_io
+        begin
+          @raw_data = JSON.parse(io.read)
+        rescue JSON::ParserError => e
+          raise Inspec::Deprecation::MalformedConfigFileError, "Could not parse deprecation config file: #{e.message}"
+        end
+
+        @groups = {}
+        @unknown_group_action = :warn
+        validate!
+      end
+
+      private
+
+      def open_default_config_io
+        default_path = File.join(Inspec.src_root, 'etc', 'deprecations.json')
+        unless File.exist?(default_path)
+          raise Inspec::Deprecation::MalformedConfigError, "Missing deprecation config file: #{default_path}"
+        end
+        File.open(default_path)
+      end
+
+      #====================================================================================================#
+      #                                          Validation
+      #====================================================================================================#
+      def validate!
+        validate_file_version
+        validate_unknown_group_action
+
+        unless @raw_data.key?('groups')
+          raise Inspec::Deprecation::InvalidConfigFileError, 'Missing groups field'
+        end
+        unless @raw_data['groups'].is_a?(Hash)
+          raise Inspec::Deprecation::InvalidConfigFileError, 'Groups field must be a Hash'
+        end
+        @raw_data['groups'].each do |group_name, group_info|
+          validate_group_entry(group_name, group_info)
+        end
+      end
+
+      def validate_file_version
+        unless @raw_data.key?('file_version')
+          raise Inspec::Deprecation::InvalidConfigFileError, 'Missing file_version field'
+        end
+        unless @raw_data['file_version'] == '1.0.0'
+          raise Inspec::Deprecation::InvalidConfigFileError, "Unrecognized file_version '#{@raw_data['file_version']}' - supported versions: 1.0.0"
+        end
+      end
+
+      def validate_unknown_group_action
+        seen = (@raw_data['unknown_group_action'] || @unknown_group_action).to_sym
+        unless VALID_ACTIONS.include?(seen)
+          raise Inspec::Deprecation::UnrecognizedActionError, "Unrecognized action for unknown groups '#{seen}' - supported actions: #{VALID_ACTIONS.map(&:to_s).join(', ')}"
```

Comment:
If I am reading it correctly I think message should be `Unrecognised action '#{seen} for unknown groups ...`

Reply:
I would like that or:

```
"Unrecognized action for unknown groups: '#{seen}' - supported actions:
```

Reply:
Or, we could match the below and do:

```
"Unrecognized action for unknown groups - saw '#{seen}', supported fields:
```

Reply:
Thank you, @Vasu1105, for pointing out that `seen` referred to an action, not a group.